internal/controller/operatorconfigmap_controller.go

```diff
@@ -323,6 +324,15 @@ func (c *OperatorConfigMapReconciler) reconcileDelegatedCSI() error {
 		return fmt.Errorf("failed to remove older csi resources: %v", err)
 	}
 
+	crd := &metav1.PartialObjectMetadata{}
+	crd.SetGroupVersionKind(extv1.SchemeGroupVersion.WithKind("CustomResourceDefinition"))
+	crd.Name = volumeGroupSnapshotClassCrd
+	if err := c.Client.Get(c.ctx, client.ObjectKeyFromObject(crd), crd); client.IgnoreNotFound(err) != nil {
+		c.log.Error(err, "Failed to get CRD", "CRD", volumeGroupSnapshotClassCrd)
+		return err
+	}
+	utils.AssertEqual(c.AvailCrds[volumeGroupSnapshotClassCrd], crd.UID != "", utils.ExitCodeThatShouldRestartTheProcess)
+
```

Comment:
This should move to storageClaim controller, where we are having a similar check for drClusterconfigs